common/app/routes/challenges/utils.js

```diff
@@ -326,12 +326,29 @@ export function filterComingSoonBetaChallenge(
 }
 
 export function filterComingSoonBetaFromEntities(
-  { challenge: challengeMap, ...rest },
+  { challenge: challengeMap, block: blocks, ...rest },
   isDev = false
 ) {
   const filter = filterComingSoonBetaChallenge.bind(null, isDev);
   return {
     ...rest,
+    block: Object.keys(blocks)
+      .map(dashedName => {
+        const block = blocks[dashedName];
+        let challenges = block.challenges;
+        if (Array.isArray(challenges)) {
+          challenges = challenges
+            .map(dashedName => challengeMap[dashedName])
+            .filter(filter)
+            .map(challenge => challenge.dashedName);
+        }
+        block.challenges = challenges;
```

Comment:
That's pretty cool! I totally didn't think about mutating state, that's really good to learn. Thanks for the tip!